exposed-tests/src/test/kotlin/org/jetbrains/exposed/sql/tests/shared/ddl/DatabaseMigrationTests.kt

```diff
@@ -257,11 +261,585 @@ class DatabaseMigrationTests : DatabaseTestsBase() {
+                }
+                TestDB.SQLSERVER -> {
+                    assertEquals(3, statements.size)
+                    assertEquals("ALTER TABLE test_table ADD NEW_id BIGINT IDENTITY(1,1)", statements[0])
+                    assertEquals("ALTER TABLE test_table DROP COLUMN id", statements[1])
+                    assertEquals("EXEC sp_rename 'test_table.NEW_id', 'id', 'COLUMN'", statements[2])
+                }
+                TestDB.ORACLE, TestDB.H2_V2_ORACLE -> {
+                    assertEquals(1, statements.size)
+                    assertEquals("CREATE SEQUENCE test_table_id_seq START WITH 1 MINVALUE 1 MAXVALUE 9223372036854775807", statements[0])
+                }
+                else -> {
+                    assertEquals(1, statements.size)
+                    val alterColumnWord = if (currentDialectTest is MysqlDialect) "MODIFY" else "ALTER"
+                    assertTrue(statements[0].startsWith("ALTER TABLE test_table $alterColumnWord COLUMN id ", ignoreCase = true))
+                }
+            }
+        }
+    }
+
+    @Test
+    fun testAddAutoIncrementWithSequenceNameToExistingColumn() {
+        val sequenceName = "custom_sequence"
+        val tableWithoutAutoIncrement = object : IdTable<Long>("test_table") {
+            override val id: Column<EntityID<Long>> = long("id").entityId()
+        }
+        val tableWithAutoIncrementSequenceName = object : IdTable<Long>("test_table") {
+            override val id: Column<EntityID<Long>> = long("id").autoIncrement(sequenceName).entityId()
+        }
+
+        withTables(excludeSettings = listOf(TestDB.SQLITE), tableWithoutAutoIncrement) {
+            assertEquals(0, MigrationUtils.statementsRequiredForDatabaseMigration(tableWithoutAutoIncrement, withLogs = false).size)
+
+            val statements = MigrationUtils.statementsRequiredForDatabaseMigration(tableWithAutoIncrementSequenceName, withLogs = false)
+            assertEquals(1, statements.size)
+            if (currentDialectTest.supportsCreateSequence) {
+                assertEquals(
+                    "CREATE SEQUENCE${" IF NOT EXISTS".takeIf { currentDialectTest.supportsIfNotExists } ?: ""} " +
+                        "$sequenceName START WITH 1 MINVALUE 1 MAXVALUE 9223372036854775807",
+                    statements[0]
+                )
+            } else {
+                val alterColumnWord = if (currentDialectTest is MysqlDialect) "MODIFY" else "ALTER"
+                assertTrue(statements[0].equals("ALTER TABLE TEST_TABLE $alterColumnWord COLUMN ID BIGINT AUTO_INCREMENT NOT NULL", ignoreCase = true))
+            }
+        }
+    }
+
+    @Test
+    fun testAddAutoIncrementWithCustomSequenceToExistingColumn() {
+        val sequence = Sequence(
+            name = "my_sequence",
+            startWith = 4,
+            incrementBy = 2,
+            minValue = 1,
+            maxValue = 100,
+            cycle = true,
+            cache = 20
+        )
```

Comment:
If this is identical across all tests and immutable, could it be a class property?